andela/bp-esa-frontend
# #164536900 Login with the andela api

**What does this PR do?**

- Ensures that the Andela API is used during login.
- Adds tests.

**Description of task to be completed?**
When a user clicks on the Login button, he is redirected to the Andela API which adds a token to the cookies. The token is used to determine user roles which are used to grant user's access. 

**How should this be manually tested?**
- Clone and set up the project using the instruction in the README.
- $ git checkout to `ft-use-andela-api-to-login-164536900` branch which contains the changes.
- Add the following variables to your frontend `.env` file: `REACT_APP_URL=http://localhost:3000/login` and `REACT_APP_ANDELA_AUTH_HOST=https://api-prod.andela.com`
- Use $ yarn start to start the application
- Start the frontend.
- Click on the login button.
You will get a `502 Bad Gateway` response.
- Open the app in an incognito window and login.
The going should be successful
- Inspect and check `cookies` under `Application`
There should be a cookie variable called `jwt-token`.
- Copy and paste the token in `https://jwt.io/` for it to be decoded.
The decoded token should include your user information which included `roles`.

**What are the relevant pivotal tracker stories?**
[#164536900](https://www.pivotaltracker.com/story/show/164536900)
